api/src/main/java/io/opentelemetry/OpenTelemetry.java

```diff
@@ -5,255 +5,185 @@
- * and {@link BaggageManager}.
+ * The entrypoint to telemetry functionality for tracing, metrics and baggage.
  *
- * <p>The telemetry objects are lazy-loaded singletons resolved via {@link ServiceLoader} mechanism.
+ * <p>The default for the OpenTelemetry API will include any {@link
+ * io.opentelemetry.trace.spi.TracerProviderFactory}, {@link
+ * io.opentelemetry.metrics.spi.MeterProviderFactory}, or {@link
+ * io.opentelemetry.baggage.spi.BaggageManagerFactory} found on the classpath, or otherwise will be
+ * default, with no-op behavior.
  *
  * @see TracerProvider
- * @see MeterProviderFactory
- * @see BaggageManagerFactory
+ * @see MeterProvider
+ * @see BaggageManager
  */
-@ThreadSafe
-public final class OpenTelemetry {
-  private static final Object mutex = new Object();
-
-  @Nullable private static volatile OpenTelemetry instance;
-
-  private final TracerProvider tracerProvider;
-  private final MeterProvider meterProvider;
-  private final BaggageManager contextManager;
-
-  private volatile ContextPropagators propagators = DefaultContextPropagators.builder().build();
+public interface OpenTelemetry {
 
-  /**
-   * Returns a singleton {@link TracerProvider}.
-   *
-   * @return registered TracerProvider or default via {@link DefaultTracerProvider#getInstance()}.
-   * @throws IllegalStateException if a specified TracerProvider (via system properties) could not
-   *     be found.
-   */
-  public static TracerProvider getTracerProvider() {
-    return getInstance().tracerProvider;
+  /** Returns the globally registered {@link TracerProvider}. */
+  static TracerProvider getTracerProvider() {
+    return DefaultOpenTelemetry.getGlobalOpenTelemetry().getMyTracerProvider();
   }
 
   /**
-   * Gets or creates a named tracer instance.
+   * Gets or creates a named tracer instance from the globally registered {@link TracerProvider}.
    *
-   * <p>This is a shortcut method for <code>getTracerProvider().get(instrumentationName)</code>.
+   * <p>This is a shortcut method for {@code getTracerProvider().get(instrumentationName)}
    *
    * @param instrumentationName The name of the instrumentation library, not the name of the
    *     instrument*ed* library (e.g., "io.opentelemetry.contrib.mongodb"). Must not be null.
    * @return a tracer instance.
    */
-  public static Tracer getTracer(String instrumentationName) {
-    return getTracerProvider().get(instrumentationName);
+  static Tracer getTracer(String instrumentationName) {
+    return DefaultOpenTelemetry.getGlobalOpenTelemetry().getMyTracer(instrumentationName);
   }
 
   /**
-   * Gets or creates a named and versioned tracer instance.
+   * Gets or creates a named and versioned tracer instance from the globally registered {@link
+   * TracerProvider}.
    *
-   * <p>This is a shortcut method for <code>
-   * getTracerProvider().get(instrumentationName, instrumentationVersion)</code>.
+   * <p>This is a shortcut method for @codegetTracerProvider().get(instrumentationName,
```

Comment:
{@code getTracreProvider()...} ?

Reply:
@anuraaga fix this up and we're ready go merge, IMO

Reply:
Ouch - thanks fixed